### PR TITLE
Add support for guzzle 6

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

**Comment by dwsupplee** [comment]
Good with me :)

**Comment by dwsupplee** [comment]
Okay with me

**Comment by bshaffer** [comment]
@tbetbetbe this is a large change, I would love another pair of eyes on it

File: .travis.yml [hunk 1]
```diff
@@ -3,18 +3,20 @@ language: php
 sudo: false
 
 php:
-  - 5.4
```
Comment (bshaffer) [inline .travis.yml]: We should keep `5.4` compatibility. Yes it's EOL'd, but only recently. And many customers still use it.

Comment (dwsupplee) [inline .travis.yml]: Guzzle 6 requires >= 5.5, I can add an exclusion for those requirements if that sounds good.

Comment (bshaffer) [inline .travis.yml]: We will still support `5.4` for Guzzle 5

**Comment by bshaffer** [comment]
@tbetbetbe @tmatsuo @sjlangley I'll be merging this by EOD unless there are any concerns

**Comment by tbetbetbe** [comment]
LGTM

File: src/Credentials/GCECredentials.php [hunk 2]
```diff
@@ -120,25 +128,37 @@ public static function onGce(ClientInterface $client = null)
...
+        'GET',
+        self::getTokenUri(),
+        [self::FLAVOR_HEADER => 'Google']
+      )
+    );
+    $body = (string) $resp->getBody();
+
+    // Assume it's JSON; if it's not throw an exception
+    if (null === $json = json_decode($body, true)) {
+      throw new \Exception('Invalid JSON response');
```
Comment (tmatsuo) [inline src/Credentials/GCECredentials.php]: Is it possible to give a hint for debugging when someone gets this exception?

File: tests/BaseTest.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  public function onlyGuzzle5()
+  {
+    $version = ClientInterface::VERSION;
+    if ('5' !== $version[0]) {
+      $this->markTestSkipped('Guzzle 5 only');
+    }
+  }
+}
```
Comment (tmatsuo) [inline tests/BaseTest.php]: nit: add newline?

**Comment by tmatsuo** [comment]
Looks good except few minor nits

File: tests/bootstrap.php [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?php
 /*
- * Copyright 2014 Google Inc.
+ * Copyright 2015 Google Inc.
```
Comment (tmatsuo) [inline tests/bootstrap.php]: I don't think you should update the year

Comment (bshaffer) [inline tests/bootstrap.php]: why not?

Comment (tmatsuo) [inline tests/bootstrap.php]: It should stay at the year when it was first written.
http://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated

Comment (bshaffer) [inline tests/bootstrap.php]: Oh! Good to know!

Well this was actually written in `2015` and I just forgot to update it when I first committed it.

Comment (tmatsuo) [inline tests/bootstrap.php]: Aha, I confirmed the first commit of this file was `2015` and then that's fine.

File: src/Credentials/GCECredentials.php (hunk 2 above)
Comment (bshaffer) [inline src/Credentials/GCECredentials.php]: yes, using [`json_last_error`](http://php.net/manual/en/function.json-last-error.php), but it's not super straightforward

**Pull request merged**